lib/elixir/lib/exception.ex

```diff
@@ -1631,3 +1631,90 @@ defmodule ErlangError do
   defp nth(3), do: "3rd"
   defp nth(n), do: "#{n}th"
 end
+
+defmodule Inspect.Error do
+  @moduledoc """
+  Raised when a struct cannot be inspected.
+  """
+  # QUESTION: I am not sure if is it considered backward incompatible
+  # to enforce keys that did not exist before
+  @enforce_keys [:exception, :stacktrace, :struct]
+  defexception exception: nil,
+               message: nil,
+               stacktrace: nil,
+               struct: nil
+
+  @impl true
+  def exception(arguments) when is_list(arguments) do
+    with {:ok, exception} <- Keyword.fetch!(arguments, :exception) |> validate(:exception),
+         {:ok, stacktrace} <- Keyword.fetch!(arguments, :stacktrace) |> validate(:stacktrace),
+         {:ok, struct} <- Keyword.fetch!(arguments, :struct) |> validate(:struct),
+         {:ok, message} <- Keyword.get(arguments, :message) |> validate(:message) do
```

Comment:
There is no reason for those validations. We should just let it crash later on if invalid. We also no longer need to handle message.

Reply:
Ok. even simpler